### PR TITLE
Create your own Enumerable

### PR DESCRIPTION
**Pull request by ahmedadel56**
- [x] Create a class MyList that has an instance variable @list.
- [x] In MyList implement a method #each that yields successive members of @list and uses the MyEnumerable module.
- [x] Create a module MyEnumerable that implements the following methods:

- #all?
- #any?
- #filter

- [x] Each class and module should have a separate .rb file.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+require './my_enumerable'
```
Comment (GzimAsani) [inline main.rb]: - [x] Use require_relative instead here because `require_relative` should be used for referring to files within your directory.

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @ahmedadel56  and @Wikitelmex                                      
Good job so far! Your code looks fine, good job :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Great job with the project it looks really good :muscle: 

## Required Changes ♻️ 
- [ ] _Check inline comments for more information :arrow_double_down:_ 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this 
You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (ahmedadel56) [reply main.rb]: done

**Comment by helman101** [comment]
# STATUS: APPROVED 🟢 🥳 

Hi @ahmedadel56 & @Wikitelmex 🥇 , 

### Congratulations!!! 🥳 🎉

You have successfully completed all the expectations for this project, there is nothing more to do here, now you can go on your way and keep improving your skills.

### Optional Changes 🤓

- [ ] [OPTIONAL] The purpose of this project is to make you understand how innumerable work, for this reason, the use of `Array.each()` is forbidden, you should create your own `each` method without using each, kindly change this method 😄

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**